docx/oxml/header.py

```diff
@@ -0,0 +1,14 @@
+from . import OxmlElement
+from .xmlchemy import BaseOxmlElement, ZeroOrMore
+
+
+class CT_Hdr(BaseOxmlElement):
+    """
+    ``<w:hdr>``, the container element for the header content
+    """
+    p = ZeroOrMore('w:p', successors=())
+
+    @classmethod
+    def new(cls):
+        header_elm = OxmlElement('w:hdr')
+        return header_elm
```

Comment:
We don't need this yet, so we should leave it out for now. But as far as the actual structure of it, if that's what you're asking, I couldn't be sure yet. I don't see us needing the `new()` classmethod, but I haven't really thought about it.

Here is something that might give an idea though, about when we create one of these:

* False is assigned to header.is_linked_to_previous
* header looks and sees whether it has a headerReference (or whatever that element is) of the right kind. Let's say it doesn't find one, which means it needs to create that header part.
* So first thing is it needs to create a new header part, then it needs to create a headerReference element to "link" to it with an rId.
* So it gets its part. That's probably a DocumentPart object, we won't care. It could be a glossary part can also have headers I don't know. All part and relationship management is done by another part. So we delegate the adding a header part to that part (definitely a DocumentPart in our case, we just won't hardwire that into the call). So the header code goes kind of like:

```python
part = self.part  # more on this later, but you can find it around in the code in places
# I'm not sure about this signature, but the gist of it will be
new_header_rId = part.add_header()
self._sectPr.add_header_reference(DEFAULT_HEADER, new_header_rId)
```

And that's all we do. We don't actually store a reference to the header, we only get that if we need it by referring to the XML and navigating the relationship etc. The XML stores the state and we don't retain a redundant copy of that state (because we don't want the job of keeping them synchronized).

Anyway, somewhere in all that a new `w:hdr` object gets created. I'm just not sure how yet. We should leave that decision until we get closer to it, probably until we're actually on the commit where we *have* to do it :) (that's the agile way :)